### DISCUSSION TITLE
How to use ohmyzsh completion plugins ?

### DISCUSSION THREAD

**Discussion by 3kezoh** [mattmc3/antidote#111] [Q&A]
Hello,

How can I use ohmyzsh completion plugins like :

https://github.com/ohmyzsh/ohmyzsh/tree/master/plugins/doctl
https://github.com/ohmyzsh/ohmyzsh/tree/master/plugins/kubectl

My `zsh_plugins.txt` file look like this :

```txt
ohmyzsh/ohmyzsh path:lib

(other oh my zsh plugins ... )

ohmyzsh/ohmyzsh path:plugins/kubectl
ohmyzsh/ohmyzsh path:plugins/doctl

(other bundles ... )

belak/zsh-utils path:completion
```

All i see in my terminal is 

```zsh
zsh: compdef: command not found...
zsh: compdef: command not found...
zsh: compdef: command not found...
zsh: compdef: command not found...
zsh: compdef: command not found...
zsh: compdef: command not found...
``` 

And when I try to get completion :

```zsh
zsh: _doctl: command not found...
zsh: _doctl: command not found...
zsh: _doctl: command not found...
```

Thank you

**Comment by 3kezoh** [comment]
I ended up adding this to my .zshrc but was wondering if there was anything better to do ?

```
# kubectl completion
source <(kubectl completion zsh)

# minikube completion
source <(minikube completion zsh)

# doctl completion
source <(doctl completion zsh)
```

**Reply by adrianbaena** [reply]
You can add `kind:fpath` next to completion plugin. It will add the file to the fpath. Thats where compinit loads completions from. Make sure you delete the .zcompdump file (or you can simply run `compinit` manually) and maybe restart the shell.

`ohmyzsh/ohmyzsh path:plugins/kubectl kind:fpath`

Will do the trick I think. I'm saying that because I have this:
![20230422_22h47m08s_grim](https://user-images.githubusercontent.com/61917364/233806348-a8887f49-7935-43cd-bc83-d6e705ba88c1.png)
In my zsh_plugins.txt and it works fine so I don't see why this wouldn't work for your case. Test it and let us know 😄

You can see if the completion script is being loaded doing: `print -rl -- $fpath`
And searching for the _(underscore)(nameoftheplugin)
Or in my case: `/home/myusername/.config/zsh/custom/completions` because i have set the whole folder for my custom completions. But in your case it will be only the file of each plugin.

I hope it helps!
